simplecpp.cpp

```diff
@@ -3126,206 +2928,182 @@ class NonExistingFilesCache {
-    return isAbsolutePath(includePath) ? absoluteSimplifiedHeaderPath : extractRelativePathFromAbsolute(absoluteSimplifiedHeaderPath);
-}
-
-static std::string openHeaderIncludePath(std::ifstream &f, const simplecpp::DUI &dui, const std::string &header)
-{
-    for (std::list<std::string>::const_iterator it = dui.includePaths.begin(); it != dui.includePaths.end(); ++it) {
-        std::string path = openHeader(f, getIncludePathFileName(*it, header));
+    // search the header on the include paths (provided by the flags "-I...")
+    for (const auto &includePath : dui.includePaths) {
+        std::string path = openHeaderDirect(f, simplecpp::simplifyPath(includePath + "/" + header));
         if (!path.empty())
             return path;
     }
     return "";
 }
 
-static std::string openHeader(std::ifstream &f, const simplecpp::DUI &dui, const std::string &sourcefile, const std::string &header, bool systemheader)
+std::pair<simplecpp::FileData *, bool> simplecpp::FileDataCache::tryload(const FileDataCache::name_map_type::iterator &name_it, const simplecpp::DUI &dui, std::vector<std::string> &filenames, simplecpp::OutputList *outputList)
 {
-    if (isAbsolutePath(header))
-        return openHeader(f, header);
+    const std::string &path = name_it->first;
+    FileID fileId;
 
-    // prefer first to search the header relatively to source file if found, when not a system header
-    if (!systemheader) {
-        std::string relativeHeader = openHeaderRelative(f, sourcefile, header);
-        if (!relativeHeader.empty()) {
-            return relativeHeader;
-        }
+    if (!getFileId(path, fileId))
+        return std::make_pair(nullptr, false);
+
+    const auto id_it = mIdMap.find(fileId);
+    if (id_it != mIdMap.end()) {
+        name_it->second = id_it->second;
+        return std::make_pair(id_it->second, false);
     }
 
-    // search the header on the include paths (provided by the flags "-I...")
-    return openHeaderIncludePath(f, dui, header);
-}
+    std::ifstream f(path);
+    FileData *const data = new FileData {path, TokenList(f, filenames, path, outputList)};
 
-static std::string findPathInMapBothRelativeAndAbsolute(const std::map<std::string, simplecpp::TokenList *> &filedata, const std::string& path) {
-    // here there are two possibilities - either we match this from absolute path or from a relative one
-    if (filedata.find(path) != filedata.end()) {// try first to respect the exact match
-        return path;
-    }
+    if (dui.removeComments)
+        data->tokens.removeComments();
 
-    // otherwise - try to use the normalize to the correct representation
-    std::string alternativePath;
-    if (isAbsolutePath(path)) {
-        alternativePath = extractRelativePathFromAbsolute(simplecpp::simplifyPath(path));
-    } else {
-        alternativePath = toAbsolutePath(path);
-    }
+    name_it->second = data;
+    mIdMap.insert(std::make_pair(fileId, data));
+    mData.push_back(std::unique_ptr<FileData>(data));
```

Comment:
Now that we have C++11 I think we can cleanup:
```suggestion
    mIdMap.emplace(fileId, data);
    mData.emplace_back(data);
```